### PR TITLE
Update `@bazel_bootlin` and enable sanitizers with GCC

### PR DESCRIPTION
**Pull request by oliverlee**
Change-Id: I760e1e7d6bd4ab43c37cf48fdfe29bd08617c261

### CODE REVIEW

File: WORKSPACE.bazel [hunk 1]
```diff
@@ -40,15 +40,16 @@ COMMON_CXX_WARNINGS = [
...
+    buildroot_version = "bleeding-edge-2022.08-1",
+    extra_cxx_flags = [
         "-std=c++23",
         "-Wduplicated-cond",
         "-Wduplicated-branches",
         "-Wlogical-op",
         "-Wuseless-cast",
         "-Wshadow=compatible-local",  # Equivalent to -Wshadow on llvm. GCC default is stupid.
     ] + COMMON_CXX_FLAGS + COMMON_CXX_WARNINGS,
+    libc_impl = "glibc",
```
Comment (garymm) [inline WORKSPACE.bazel:52]: surprised this isn't the default

**Review by garymm — APPROVED** [review]

File: WORKSPACE.bazel (hunk 1 above)
Comment (oliverlee) [reply WORKSPACE.bazel:52]: Want to submit a PR to make it the default?

Comment (garymm) [reply WORKSPACE.bazel:52]: I don't have enough context to know if that's right. We can talk about it next time we meet up.

Comment (oliverlee) [reply WORKSPACE.bazel:52]: Note for future conversation.

The attribute is introduced here:
https://github.com/oliverlee/bazel_bootlin/blob/main/defs.bzl#L110

and it is used to specify the toolchain to download (glibc, musl, uclibc).

**Pull request merged**